framework/platform/platform.cpp

```diff
@@ -185,49 +177,7 @@ ExitCode Platform::main_loop()
 
 	while (!window->should_close() && !close_requested)
 	{
-		try
-		{
-			// Load the requested app
-			if (app_requested())
-			{
-				if (!start_app())
-				{
-					throw std::runtime_error("Failed to load requested application");
-				}
-
-				// Compensate for load times of the app by rendering the first frame pre-emptively
-				timer.tick<Timer::Seconds>();
-				active_app->update(0.01667f);
-			}
-
-			update();
-
-			if (active_app && active_app->should_close())
-			{
-				std::string id = active_app->get_name();
-				on_app_close(id);
-				active_app->finish();
-			}
-
-			window->process_events();
-		}
-		catch (std::exception &e)
-		{
-			LOGE("Error Message: {}", e.what());
-			LOGE("Failed when running application {}", active_app->get_name());
-
-			on_app_error(active_app->get_name());
-
-			if (app_requested())
-			{
-				LOGI("Attempting to load next application");
-			}
-			else
-			{
-				set_last_error(e.what());
-				return ExitCode::FatalError;
-			}
-		}
+		main_loop_frame();
```

Comment:
You're missing any failure exit code from main_loop_frame.
Maybe something like
```
ExitCode Platform::main_loop()
{
	ExitCode exit_code = ExitCode::Success;
	if (!app_requested())
	{
		exit_code = ExitCode::NoSample;
	}

	while ((exit_code == ExitCode::Success) && !window->should_close() && !close_requested)
	{
		exit_code = main_loop_frame();
	}

	return exit_code;
}

```

Reply:
D'oh! That's my bad. Thanks for catching it. Fixed now.

I've left the initial `!app_requested()` test as an immediate return, to clarify it's use as a one-time pre-check, and keep it disentangled from the looping logic.